src/Microsoft.OpenApi.OData.Reader/Operation/EdmOperationOperationHandler.cs

```diff
@@ -146,6 +146,31 @@ protected override void SetTags(OpenApiOperation operation)
             base.SetTags(operation);
         }
 
+        /// <summary>
+        /// Genrates the tag name for the operation.
+        /// </summary>
+        /// <param name="tagName">The generated tag name.</param>
+        /// <param name="skip">The number of segments to skip.</param>
+        private void GenerateTagName(out string tagName, int skip = 1)
+        {            
+            var targetSegment = Path.Segments.Reverse().Skip(skip).First();
+
+            if (targetSegment is ODataNavigationPropertySegment)
+            {
+                tagName = EdmModelHelper.GenerateNavigationPropertyPathTagName(Path, Context);
+            }
+            else if (targetSegment is ODataOperationSegment || targetSegment is ODataOperationImportSegment // Composable function
+                || targetSegment is ODataKeySegment) // Previous segmment could be a navigation property or a navigation source segment
+            {   
+                skip += 1;
+                GenerateTagName(out tagName, skip);
+            }
+            else // ODataNavigationSourceSegment
+            {
+                tagName = NavigationSource.Name + "." + NavigationSource.EntityType.Name;
+            }

```

Comment:
Just a suggestion. This could be rewritten as a switch statement as below.

```suggestion
            switch (targetSegment)
            {
                case ODataNavigationPropertySegment:
                    tagName = EdmModelHelper.GenerateNavigationPropertyPathTagName(Path, Context);
                    break;
                case ODataOperationSegment:
                case ODataOperationImportSegment:
                // Previous segmment could be a navigation property or a navigation source segment
                case ODataKeySegment:
                    skip += 1;
                    GenerateTagName(out tagName, skip);
                    break;
                // ODataNavigationSourceSegment
                default:
                    tagName = NavigationSource.Name + "." + NavigationSource.EntityType.Name;
                    break;
            }
```